pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -185,6 +191,58 @@ public void testInvalidTableConfig() {
     }
   }
 
+  @Test
+  public void testRefreshTableConfigAndQueryTimeout()
+      throws Exception {
+    TableConfig tableConfig = _helixResourceManager.getOfflineTableConfig(getTableName());
+    assertNotNull(tableConfig);
+
+    // Set timeout as 5ms so that query will timeout
+    tableConfig.setQueryConfig(new QueryConfig(5L));
+    _helixResourceManager.updateTableConfig(tableConfig);
+
+    // Wait for at most 1 minute for broker to receive and process the table config refresh message
+    TestUtils.waitForCondition(aVoid -> {
```

Comment:
seems like a candidate for intermittent failure, good to add a comment on the circumstances under which this test can fail. I cant think of another way to add the test

Reply:
I removed the test on uncertain properties (such as `timeUsedMs`), and it should always pass right not.